crates/holochain/tests/tests/schedule.rs

```diff
@@ -0,0 +1,286 @@
+            Some(Schedule::Ephemeral(duration)) => duration.as_millis() as u64,
+            None => 3,
+            _ => panic!("Expected Ephemeral Schedule"),
+        };
+        match ms {
+            0 => Ok(None),
+            _ => Ok(Some(Schedule::Ephemeral(std::time::Duration::from_millis(
+                ms - 1,
+            )))),
+        }
+    });
+
+    let dna = SweetDnaFile::unique_from_inline_zomes(zome).await;
+    let mut conductor = SweetConductor::from_standard_config().await;
+    let app = conductor.setup_app("app", &[dna.0.clone()]).await.unwrap();
+    let cell = app.into_cells()[0].clone();
+    let mut app_signal = conductor.subscribe_to_app_signals("app".into());
+    let host_fn_caller =
+        HostFnCaller::create_for_zome(cell.cell_id(), &conductor.raw_handle(), &dna.0, 0).await;
+
+    // Start test: schedule function
+    conductor
+        .call::<(), ()>(&cell.zome("coordinator"), "start", ())
+        .await;
+    // Scheduled function is first called with None input.
+    // Then should be called with decreasing durations until it gets unscheduled.
+    assert_eq!(None, wait_for_signal(&mut app_signal).await.unwrap());
+    assert_eq!(
+        Some(Schedule::Ephemeral(std::time::Duration::from_millis(2))),
+        wait_for_signal(&mut app_signal).await.unwrap()
+    );
+    assert_eq!(
+        Some(Schedule::Ephemeral(std::time::Duration::from_millis(1))),
+        wait_for_signal(&mut app_signal).await.unwrap()
+    );
+    assert_eq!(
+        Some(Schedule::Ephemeral(std::time::Duration::from_millis(0))),
+        wait_for_signal(&mut app_signal).await.unwrap()
+    );
+    assert!(wait_for_signal(&mut app_signal).await.is_err());
+    assert!(!is_scheduled(&host_fn_caller, &cell).await);
+}
+
+/// Test persisted schedule with invalid crontab output
+/// Assuming a schedular interval of 100ms
+#[tokio::test(flavor = "multi_thread")]
+async fn schedule_bad_crontab() {
+    holochain_trace::test_run();
+
+    // Start with a valid crontab then set an invalid one.
+    let zome = create_schedule_zome(|api, input| {
+        let _ = api.emit_signal(AppSignal::new(ExternIO::encode(input.clone()).unwrap()));
+        match input {
+            None => Ok(Some(Schedule::Persisted("*/1 * * * * * *".to_string()))), // every second
+            Some(_) => Ok(Some(Schedule::Persisted("*/0 * * * * * *".to_string()))), // invalid crontab
+        }
+    });
+
+    let dna = SweetDnaFile::unique_from_inline_zomes(zome).await;
+    let mut conductor = SweetConductor::from_standard_config().await;
+    let app = conductor.setup_app("app", &[dna.0]).await.unwrap();
+    let cell = app.into_cells()[0].clone();
+    let mut app_signal = conductor.subscribe_to_app_signals("app".into());
+
+    conductor
+        .call::<(), ()>(&cell.zome("coordinator"), "start", ())
+        .await;
+
+    // Scheduled function is first called with None input
+    assert_eq!(None, wait_for_signal(&mut app_signal).await.unwrap());
```

Comment:
```suggestion
    assert!(wait_for_signal(&mut app_signal).await.is_none());
```

Reply:
That is more compact and I was wondering about it above but actually it's nice to see what's inside the `Some` if this fails